phoebe/atmospheres/models.py

```diff
@@ -0,0 +1,506 @@
+        multiplying with this factor.
+
+    In addition, the following methods need to or may be overloaded:
+
+    * `parse_rules`: provides rules for parsing atmosphere fits filenames to
+        extract basic axis values.
+    * `limb_treatment`: defines how intensities at the exact limb (mu=0) should
+        be treated. By default, the intensities are linearly extrapolated to
+        mu=0.
+
+    When a model atmosphere is instantiated via the from_path() method, the basic
+    axes are populated with unique values from the filenames of the atmosphere
+    fits files. The axes are then exported as numpy arrays.
+
+    Attributes that are automatically populated:
+
+    * `basic_axes` (tuple): tuple of numpy arrays for basic axes
+
+    In the case of from_path() instantiation, the following attributes are
+    also populated:
+
+    * `models` (list): list of atmosphere fits files
+    * `nmodels` (int): number of atmosphere fits files
+    * `indices` (array): array of indices for all defined nodes in the model
+        atmosphere
+    * `[axis_name]` (array): numpy array for each axis, where the name is
+        automatically inferred from the basic axis names
+
+    Arguments
+    ----------
+    * `basic_axes` (tuple of ndarrays): values of the basic axes
+    * `from_path` (bool): if True, the class is instantiated from a path
+
+    Raises
+    -------
+    * `FileNotFoundError`: if the path does not exist
+    """
+
+    name = None
+    prefix = None
+    external = False
+
+    # default axes:
+    basic_axis_names = ['teffs', 'loggs', 'abuns']
+
+    def __init__(self, basic_axes=None, from_path=False):
+        if from_path:
+            return
+
+        self.path = None
+
+        # the model needs to either provide tabulated intensities or
+        # a function to compute them. If the model provides tabulated
+        # intensities, basic_axes must be defined; if it provides a
+        # function, basic_axes will be automatically determined from
+        # axis definitions via class attributes.
+
+        if hasattr(self, 'intensity') and callable(self.intensity):
+            # the model provides a function to compute intensities.
+            for axis_name in self.basic_axis_names:
+                if not hasattr(self, axis_name):
+                    raise ValueError(f'Model atmosphere named basic axis "{axis_name}" but it did not define it.')
+            basic_axes = tuple([getattr(self, axis_name) for axis_name in self.basic_axis_names])
+        else:
+            if basic_axes is None:
+                raise ValueError('basic_axes must be defined.')
+
+        self.basic_axes = basic_axes
+
+    def __repr__(self):
+        return self.name
+
+    def __str__(self):
+        return self.name
+
+    def register(self):
+        """
+        Registers the model atmosphere with the global model table.
+        """
+
+        global _atmtable
+        _atmtable.append(self.__class__)
```

Comment:
let's eventually replace this with a proper registry system on the classes (this does the trick for now though... just wanted to make a note)

Reply:
actually, is this used anywhere?  Or was it replaced by the `__subclasses__()` in [line 486](https://github.com/phoebe-project/phoebe2/pull/1007/files#diff-d6013f3dea32bb3d8c28c4af4a86c0944d5a7765f9f87fb98c3d2ff9ee174f9cR486)?